### PR TITLE
RHOAIENG-8554: Add static-checks GitHub action for go code

### PR DESCRIPTION
**Pull request by jiridanek**
https://issues.redhat.com/browse/RHOAIENG-8554

## Description
Adds static-checks GitHub action for go code

## How Has This Been Tested?
See the GHA run

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/notebook-controller/.golang-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# https://github.com/golangci/golangci-lint
```
Comment (atheo89) [inline components/notebook-controller/.golang-ci.yaml:1]: I would propose to add this file to a dedicated folder for CIs if possible. (The same for odh)

**Review by atheo89 — COMMENTED** [review]
I put a small suggestion. 
Overall it looks good to me

File: components/notebook-controller/.golang-ci.yaml (hunk 1 above)
Comment (jiridanek) [reply components/notebook-controller/.golang-ci.yaml:1]: Not possible ;( https://golangci-lint.run/usage/configuration/#config-file

Comment (atheo89) [reply components/notebook-controller/.golang-ci.yaml:1]: Ahh ok!

**Comment by jstourac** [comment]
LGTM, thanks!

The question for the vulnerability check - it is failing right now, do we intend to merge this as is even it's failing? I guess the answer is yes?

**Comment by jiridanek** [comment]
> The question for the vulnerability check - it is failing right now, do we intend to merge this as is even it's failing? I guess the answer is yes?

I'd merge it failing, and then would try to quickly fix it. It should not be too hard, I hope.

The govulncheck is less sensitive than snyk, it analyzes whether the vulnerable function is called and only reports of that is the case. So it should not report too many problems.

If I manage to improve the integration test to actually spawn a notebook in the clusters then we would be able to merge dependency updates when the test passes. No other manual work would be necessary to verify per PR.

**Pull request merged**